components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -401,36 +401,37 @@ SUBROUTINE p3_main_pre_main_loop(kts, kte, kbot, ktop, kdir, log_predictNc, dt,
        inv_rho(k) = 1._rtype/rho(k)
        qvs(k)     = qv_sat(t(k),pres(k),0)
        qvi(k)     = qv_sat(t(k),pres(k),1)
-
-       sup(k)     = qv(k)/qvs(k)-1._rtype
+                        
        supi(k)    = qv(k)/qvi(k)-1._rtype
 
        rhofacr(k) = bfb_pow(rhosur*inv_rho(k), 0.54_rtype)
        rhofaci(k) = bfb_pow(rhosui*inv_rho(k), 0.54_rtype)
        dum        = 1.496e-6_rtype * bfb_pow(t(k), 1.5_rtype) / (t(k)+120._rtype)  ! this is mu
        acn(k)     = g*rhow/(18._rtype*dum)  ! 'a' parameter for droplet fallspeed (Stokes' law)
 
-       !specify cloud droplet number (for 1-moment version)
-       if (.not.(log_predictNc)) then
-          nc(k) = nccnst*inv_rho(k)
-       endif
 
```

Comment:
I think choosing to not include ncnuc in log_nucleationPossible is ultimately the right thing to do, but the reasoning is subtle. Maybe add a comment just above the line below saying "!log_nucleationPossible is used in combination with log_hydrometeorsPresent to decide that nothing microphysical is happening so the step can be skipped. We don't include ncnuc in this determination because macrophysics is supposed to handle the condensation associated with new droplets; if it says qc=0 (as implied by log_hydrometeorsPresent=.false.), then nc will be set to zero regardless of what ncnuc says." Does this comment make sense to other folks?